packages/flutter_tools/lib/src/web/file_generators/flutter_js.dart

```diff
@@ -8,24 +8,141 @@
 /// flutter.js should be completely static, so **do not use any parameter or
 /// environment variable to generate this file**.
 String generateFlutterJsFile() {
-  return '''
+  return r'''
 // Copyright 2014 The Flutter Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-/**
- * This script installs service_worker.js to provide PWA functionality to
- *     application. For more information, see:
- *     https://developers.google.com/web/fundamentals/primers/service-workers
- */
-
 if (!_flutter) {
   var _flutter = {};
 }
 _flutter.loader = null;
 
 (function() {
   "use strict";
+  /**
+   * Wraps `promise` in a timeout of the given `duration` in ms.
+   *
+   * Resolves/rejects with whatever the original `promises` does, or rejects
+   * if `promise` takes longer to complete than `duration`. In that case,
+   * `debugName` is used to compose a legible error message.
+   *
+   * If `duration` is <= 0, the original `promise` is returned unchanged.
```

Comment:
This behavior seems a little odd to me, as it's kind of the opposite of what I would expect the behavior to be. As in, I would expect a value of zero to aggressively time out unless the promise is already resolved. Is there a reason we'd want this behavior?

Reply:
Yep, I can't think of a case to aggressively timeout something like that, but it might be usable to guarantee that a promise is resolved synchronously (maybe). Let me do this < 0 instead <= 0.